qa/integration/specs/monitoring_api_spec.rb

```diff
@@ -286,6 +287,46 @@
     end
   end
 
+  it "should retrieve plugin level flow metrics" do
+    logstash_service = @fixture.get_service("logstash")
+    logstash_service.start_with_stdin(plugins_config)
+    logstash_service.wait_for_logstash
+    number_of_events.times {
+      logstash_service.write_to_stdin("Testing plugin-level flow metrics")
+      sleep(1)
+    }
+
+    Stud.try(max_retry.times, [StandardError, RSpec::Expectations::ExpectationNotMetError]) do
+      # node_stats can fail if the stats subsystem isn't ready
+      result = logstash_service.monitoring_api.node_stats rescue nil
+      expect(result).not_to be_nil
+      # we use fetch here since we want failed fetches to raise an exception
+      # and trigger the retry block
```

Comment:
Comment seems to reference using `Hash#fetch` to get a `KeyError` to trigger a retry. Let's re-word it to match what we are actually doing:

```suggestion
      # if the result is nil, we probably aren't ready yet
      # our assertion failure will cause Stud to retry
      expect(result).not_to be_nil

```